app/desktop/studio_server/tool_servers_api.py

```diff
@@ -81,18 +81,23 @@ async def get_available_tools(
         available_tools = []
         for server in project.external_tool_servers(readonly=True):
             # Get available tools from remote MCP servers only
-            if server.type == ToolServerType.remote_mcp:
-                async with MCPSessionManager.shared().mcp_client(server) as session:
-                    tools_result = await session.list_tools()
-                    available_tools.extend(
-                        [
-                            ToolApiDescription(
-                                id=f"{MCP_REMOTE_TOOL_ID_PREFIX}{server.id}::{tool.name}",
-                                name=tool.name,
-                                description=tool.description,
-                            )
-                            for tool in tools_result.tools
-                        ]
+            match server.type:
+                case ToolServerType.remote_mcp:
+                    async with MCPSessionManager.shared().mcp_client(server) as session:
+                        tools_result = await session.list_tools()
+                        available_tools.extend(
+                            [
+                                ToolApiDescription(
+                                    id=f"{MCP_REMOTE_TOOL_ID_PREFIX}{server.id}::{tool.name}",
+                                    name=tool.name,
+                                    description=tool.description,
+                                )
+                                for tool in tools_result.tools
+                            ]
+                        )
+                case _:
+                    raise RuntimeError(
```

Comment:
this isn't an exhaustive check, just a runtime error. The type checker won't catch it, we'll still miss it if we add a value. Look at `raise_exhaustive_enum_error`